src/imap.rs

```diff
@@ -45,6 +45,7 @@ pub struct Imap {
     session: Arc<Mutex<Option<Session>>>,
     connected: Arc<Mutex<bool>>,
     interrupt: Arc<Mutex<Option<stop_token::StopSource>>>,
+    skip_next_idle_wait: Arc<Mutex<bool>>,
```

Comment:
not blocking, but should be fixed before merging to master: `Arc<Mutex<bool>>` should really be an `AtomicBool` instead